File: Euler_4/Euler_4.swift
var total: Int = 0
var a: Int = 999
var b: Int = 999


func isPalindrome(n: Int) -> Bool
{
  var result: Int = 0
  var remainder: Int
  var change: Int = n

  while change > 0{
    remainder = change % 10
    change /= 10
    result = result * 10 + remainder
  }

  if result == n{
    return true
  }else{
    return false
  }

}

while a >= 900
{
  total = a * b
  if isPalindrome(n: total){
    break
  }
  a -= 1
  if a == 900{
    a = 999
    b -= 1
  }
}
print (total)

File: Euler_004/Euler_4.go
package main

import "fmt"

func main() {
	total, a, b := 0, 999, 999

	for a >= 900 {
		total = a * b
		if reverseDigits(total) {
			break
		}
		a--
		if a == 900 {
			a = 999
			b--
		}
	}
	fmt.Println(total)
}

func reverseDigits(reverseThis int) bool {
	result, initial := 0, reverseThis

	for reverseThis > 0 {
		remainder := reverseThis % 10
		reverseThis /= 10
		result = result*10 + remainder
	}

	if result == initial {
		return true
	} else {
		return false
	}
}
